### ISSUE TITLE
Edge-like vertical tab bar or Tree-like vertical tab bar

### ISSUE THREAD

**Issue by Marin-Kitagawa** [BrowserWorks/waterfox#3042]
**Is your feature request related to a problem? Please describe.**
Waterfox supports tab bar to moved top or bottom (and relative to status and address bar) and also supports autohiding of tab bar. But could you please support for Tree-like tabs (please check my repo [FirefoxCSS](https://github.com/Marin-Kitagawa/FirefoxCSS) where I use Tree style tabs) or at least Edge-like vertical tabs?

**Describe the solution you'd like**
Vertical tabs which either has Tree-like tabs or Edge-like tabs

**Describe alternatives you've considered**
I very well can use Firefox with my CSS above with the extension. But Firefox doesn't have all the features that Waterfox has. Waterfox considers user requests and have added a lot of features. So, I feel WF will add this as well

**Comment by benbohmer** [comment]
@Marin-Kitagawa there are extension that do this very well, with a lot more features than Edge has as well.
Sidebery is great, Tree Style Tab is extremely customisable.

**Comment by Marin-Kitagawa** [comment]
@benbohmer Yes. I already mentioned that in my comment as well. I was mentioning about native support

**Comment by christmasjoke** [comment]
these extensions dont really work because the tabs are still on top too while in edge the tabs are only on the side or only on top

**Comment by MrAlex94** [comment]
Hi all!

Yes, this has been on the roadmap for a while now. Will keep everyone posted, as and when :-)

**Comment by Marin-Kitagawa** [comment]
> these extensions dont really work because the tabs are still on top too while in edge the tabs are only on the side or only on top

@christmasjoke Yes. That's exactly my point as well

**Comment by Marin-Kitagawa** [comment]
> @Marin-Kitagawa there are extension that do this very well, with a lot more features than Edge has as well. Sidebery is great, Tree Style Tab is extremely customisable.

Besides, there is a browser called `Floorp`. This does what you said. Just using CSS to "hide" the tab bar doesn't work. It is still loads when the app loads. This browser (`Floorp`) doesn't provide native vertical tab support but comes with the TST extension

**Comment by MrAlex94** [comment]
Indeed, I want it to be a native, simplistic version that looks good. Nothing overly complicated (at least not in the short term). I’m speaking with designers for the moment and hashing out some concepts - so will keep everyone updated in due course.

**Comment by Sa-Ja-Di** [comment]
> Indeed, I want it to be a native, simplistic version that looks good. Nothing overly complicated (at least not in the short term). I’m speaking with designers for the moment and hashing out some concepts - so will keep everyone updated in due course.

@MrAlex94 Native implementations are always better than over extensions - i hope you are able to add this - right now considering what for a backup browser i will use beside Pale Moon for problematic sides.

Testing right now out Waterfox and Floorp - if Waterfox would add native side-tabs - i guess then i could make Waterfox my backup one.

**Comment by timur-g** [comment]
Is this Sidebar Tab? If so, should it be closed as that is available from 6.0?

**Comment by IRainman** [comment]
Now it can be merged from Firefox.

**Comment by JSGRD22** [comment]
I feel this can be closed now with the integration of tree style tabs into the native firefox vertical tabs
